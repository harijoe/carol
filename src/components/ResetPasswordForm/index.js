import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import injectTranslate from 'i18n/hoc/injectTranslate'
import { theme, breakpoint } from 'utils/style'

import Button from 'components/Button'
import CarouselPageTemplate from 'templates/CarouselPageTemplate'
import AnimatedPasswordField from 'components/AnimatedPasswordField'

const Form = styled.form`
  width: 100%;
  box-sizing: border-box;

  ${breakpoint('m')`
    width: 100%;
  `} strong {
    color: ${theme('colors.danger')};
  }
`

const StyledButton = styled(Button)`
  margin-top: ${theme('spaces.xl')};

  ${breakpoint('m')`
    margin-top: ${theme('spaces.xxl')};
  `}
`

const ResetPasswordForm = ({ error, handleSubmit, loading, translate }) =>
  <CarouselPageTemplate heading={translate('user.reset_password.heading')} description={translate('user.reset_password.description')}>
    <Form onSubmit={handleSubmit}>
      <Field name="token" type="hidden" component="input" />
      {error && <FormattedMessage id={error} tagName="strong" />}
      <AnimatedPasswordField name="password" label={translate('user.reset_password.password')} />
      <AnimatedPasswordField name="passwordValidation" label={translate('user.confirm_password')} toggleDisabled />
      <StyledButton type="submit" loading={loading}>
        <FormattedMessage id="user.send" />
      </StyledButton>
    </Form>
  </CarouselPageTemplate>

ResetPasswordForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.string,
  translate: PropTypes.func.isRequired,
}

export default injectTranslate(ResetPasswordForm)
