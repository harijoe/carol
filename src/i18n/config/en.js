export default {
  url: 'https://carol-co-uk-dev.qarx.io',
  countryLabel: 'country.great_britain',
  contentSiteUrl: 'https://www.quotatis.co.uk/advice/',
  mainMenu: {
    resource: {
      guide: { path: 'guide', target: '_blank' },
      inspiring: { path: 'inspiration', target: '_blank' },
      faq: { path: 'FAQ', target: '_blank' },
    },
    directory: '/tradesmen-directory',
    proUrl: 'https://www.quotatispro.co.uk',
  },
  genericProjectImages: [
    '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1500641857/FR/ChatbotImages/Q2/electricite-domotique-alarmes/installation-electrique-de-l-habitat.jpg',
    '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1500641791/FR/ChatbotImages/Q2/construction-extension-renovation/renovation-de-maison-appartement-commerce.jpg',
    '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1500641950/FR/ChatbotImages/Q2/experts-et-conseils-avant-travaux/architectev2.jpg',
    '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1500641788/FR/ChatbotImages/Q2/construction-extension-renovation/amenagement-de-combles.jpg',
    '//res.cloudinary.com/quotatis/image/upload/c_fill,g_custom:adv_faces,h_300,w_200/v1505381549/FrontApp/ChatbotImages/shutterstock_408737758.jpg',
    '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1505381593/FrontApp/ChatbotImages/shutterstock_536591941.jpg',
  ],
  genericFirmDetailImage: 'http://res.cloudinary.com/quotatis/image/upload/v1508320474/FrontApp/proprofile-cover-image.jpg',
  searchSuggestions: {
    primary: [
      {
        title: 'Conservatories and orangeries',
        slug: 'conservatories-and-orangeries',
        imageUrl:
          '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1498836491/GB/ChatbotImages/Q1/conservatories-roofing-and-extensions.jpg',
        tag: 'Project',
      },
      {
        title: 'Boilers',
        slug: 'boilers',
        imageUrl:
          '//res.cloudinary.com/quotatis/image/upload/c_fill,g_west,h_300,w_200/v1498836890/GB/ChatbotImages/Q1/heating-and-air-conditioning.jpg',
        tag: 'Project',
      },
      {
        title: 'Windows',
        slug: 'windows',
        imageUrl: '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1498837780/GB/ChatbotImages/Q1/windows-and-doors.jpg',
        tag: 'Project',
      },
      {
        title: 'Solar panels',
        slug: 'solar-panels',
        imageUrl:
          '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1498836105/GB/ChatbotImages/Q1/electrical-cctv-and-solar.jpg',
        tag: 'Project',
      },
      {
        title: 'Guttering soffits and fascias',
        slug: 'guttering-soffits-and-fascias',
        imageUrl: '//res.cloudinary.com/quotatis/image/upload/c_fill,h_300,w_200/v1495120786/GB/ChatbotImages/Q1/exterior-renovation.jpg',
        tag: 'Project',
      },
    ],
    secondary: [
      {
        title: 'Roofing',
        slug: 'roofing',
      },
      {
        title: 'Paving and Driveways',
        slug: 'paving-and-driveways',
      },
      {
        title: 'Asbestos',
        slug: 'asbestos',
      },
    ],
  },
  footer: {
    advices: {
      column_1_1: {
        key: 'solar_pv',
        src: 'guide/solar-panels/solar-pv/',
      },
      column_1_2: {
        key: 'windows',
        src: 'guide/windows-doors/windows/',
      },
      column_1_3: {
        key: 'conservatories',
        src: 'guide/conservatories-roofing-extensions/conservatories/',
      },
      column_2_1: {
        key: 'gas_boiler',
        src: 'guide/heating-air-con/gas-boilers/',
      },
      column_2_2: {
        key: 'pitched_roofing',
        src: 'guide/conservatories-roofing-extensions/pitched-roofing/',
      },
      column_2_3: {
        key: 'fascias',
        src: 'guide/conservatories-roofing-extensions/soffits-fascias/',
      },
      column_3_1: {
        key: 'driveways',
        src: 'guide/paving-driveways/driveways/',
      },
      column_3_2: {
        key: 'asbestos',
        src: 'guide/asbestos/asbestos-removal/',
      },
      column_3_3: {
        key: 'guttering',
        src: 'guide/conservatories-roofing-extensions/guttering/',
      },
      column_4_1: {
        key: 'tree_surgery',
        src: 'guide/garden-maintenance/tree-surgery/',
      },
      column_4_2: {
        key: 'air-con',
        src: 'guide/heating-air-con-insulation/air-con/',
      },
      column_4_3: {
        key: 'satellite',
        src: 'guide/electrical-cctv/aerials-satellite/',
      },
    },
    corporate: {
      column_1_1: {
        key: 'create_a_job',
        clickOnFindAPro: true,
      },
      column_1_2: {
        key: 'trades',
        src: 'home-improvement-quotes',
        target: '_blank',
      },
      column_1_3: {
        key: 'about',
        src: 'who-we-are',
        contentSite: true,
        forceRedirect: true,
      },
      column_1_4: {
        key: 'contact',
        src: '/contact.html',
        target: '_blank',
      },
      column_1_5: {
        key: 'use_conditions',
        src: 'privacy-policy',
        contentSite: true,
        forceRedirect: true,
      },
      column_1_6: {
        key: 'sitemap',
        src: '/sitemap',
        target: '_blank',
      },
      column_2_1: {
        key: 'news',
        src: 'http://blog.quotatis.co.uk',
        target: '_blank',
      },
      column_2_2: {
        key: 'guides',
        src: '',
        contentSite: true,
        forceRedirect: true,
        target: '_blank',
      },
      column_2_3: {
        key: 'directory',
        src: 'tradesmen-directory',
        forceRedirect: true,
        target: '_blank',
      },
      column_2_4: {
        key: 'refer_a_pro',
        src: 'proreferral?spn=342',
        forceRedirect: true,
        target: '_blank',
      },
      column_2_5: {
        key: 'trade_associations',
        src: 'trade-associations',
        forceRedirect: true,
        target: '_blank',
      },
      column_2_6: {
        key: 'qpro',
        src: 'https://www.quotatispro.co.uk/register.html?spn=103',
        forceRedirect: true,
        target: '_blank',
      },
      column_2_7: {
        key: 'affiliates',
        src: 'https://affiliate.quotatis.co.uk/?spn=103',
        forceRedirect: true,
        target: '_blank',
      },
      column_2_8: {
        key: 'adeo',
        src: 'https://www.adeo.com/en/business-areas/chains/quotatis',
        forceRedirect: true,
        target: '_blank',
      },
    },
    social: {
      facebook: 'https://www.facebook.com/Quotatis',
      linkedin: 'https://www.linkedin.com/company/quotatisuk',
      twitter: 'https://twitter.com/Quotatis',
      google: 'https://plus.google.com/106376832204465138273',
      youtube: 'https://www.youtube.com/user/quotatisuk',
      pinterest: 'https://www.pinterest.com/quotatis/',
    },
  },
}
