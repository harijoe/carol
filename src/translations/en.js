const en = {
  home: 'Home',
  directory: 'Directory artisans EN',
  help: 'Help',
  login: 'Login',
  burger_menu: 'Burger Menu',
  server_error: 'An error occured, please try again later',
  search: 'Search',
  construction: 'Construction',
  renovation: 'Renovation',
  fixing: 'Fixing',
  choose: 'Choose',
  'Invalid username and password combination': 'Invalid username and password combination',
  'Missing parameters. "username" and "password" required': 'Missing parameters. "username" and "password" required',
  'Phone number is invalid': 'Phone number is invalid',
  'Verification code is incorrect': 'The validation code is invalid',
  'mobile_phone.invalid': 'Phone number is not valid',

  'country.france': 'France',
  'country.spain': 'Spain',
  'country.great_britain': 'Great-Britain',

  'main.baseline': 'Vos travaux en mieux EN',

  'firm.find_pro': 'Find a pro',
  'firm.resource': 'Advice',
  'firm.i_am_pro.one': 'You are',
  'firm.i_am_pro.two': 'a pro?',
  'firm.all': 'Tous nos conseils EN',
  'firm.guide': 'Guides Travaux EN',
  'firm.and_you': 'Les travaux et vous EN',
  'firm.inspiring': 'Inspirations EN',
  'firm.faq': 'FAQ EN',
  'firm.good_news': 'Good news',
  'firm.result_description': 'artisans ont été trouvés. Quotatis va maintenant selectionner jusqu\'à 5 pros selon leur spécialisation, localisation et disponibilité. EN',
  'firm.site_search': 'Site search',
  'firm.site_search.result_link': 'More details',
  'firm.details.logo': 'Logo',
  'firm.details.logo_missing': 'No logo',
  'firm.details.name': 'Firm Name',
  'firm.details.postal_code': 'Postal code',
  'firm.details.city': 'City',
  'firm.details.country': 'Country',
  'firm.details.registration_number': 'Registration number',
  'firm.details.description': 'Description',
  'firm.details.website': 'Website',
  'firm.details.website_missing': 'No website',
  'firm.details.employees_number': 'Number of employee',
  'firm.details.client_since': 'Member since',
  'firm.details.home_improvement_id': 'Home Improvement Id',
  'firm.details.certificate': 'Certificate',
  'firm.details.budget': 'Budget',
  'firm.details.delay': 'Delay',
  'firm.details.conviviality': 'Conviviality',
  'firm.details.notation': 'Notation',
  'firm.details.served_area_cities': 'Served Area Postal Codes',
  'firm.details.city_code': 'City code',
  'firm.details.reviews': 'review',

  'project.validation_page': 'Page Validation EN',
  'project.validation.describe': 'Pour nous permettre de sélectionner les professionnels de confiance les plus adaptés à votre projet, veuillez renseigner les informations ci-dessous. EN',
  'project.submit_button': 'OK EN',
  'project.submit_success_title': 'Félicitations! EN',
  'project.submit_failed_title': 'Oups! EN',
  'project.submit_success': 'Projet envoyé! EN',
  'project.submit_failed': 'Projet n\'a pas été envoyé! EN',
  'project.created_at': 'Created at',
  'project.updated_at': 'Updated at',
  'project.status': 'Status',
  'project.status.completion_in_progress': 'Projet en cours d’élaboration EN',
  'project.status.to_validate': 'Projet élaboré mais user non auto-validé EN',
  'project.status.pending_search': 'Projet en cours de recherche de pros : pas de pro trouvé EN',
  'project.status.found': 'Projet en cours de recherche de pros : X PRO(S) TROUVÉS EN',
  'project.status.started': 'Projet démarré entre un artisan et un particulier EN',
  'project.status.finished': 'Projet terminé EN',
  'project.describe.completion_in_progress': 'Vous n\'avez pas encore répondu à toutes les questions permettant de valider votre projet. EN',
  'project.describe.to_validate': 'Pour finaliser votre projet et vous envoyer une liste de professionnels de confiance, vous devez finaliser votre projet. EN',
  'project.describe.pending_search': 'Nous avons lancé la recherche de professionnels de confiance qui pourront répondre à votre projet. Vous recevrez leurs coordonnées par email et très prochainement vous pourrez directement consulter leurs fiches détaillées dans cet espace. EN',
  'project.describe.finished': 'fnished EN',
  'project.button.in_progress': 'Continuer mon projet EN',
  'project.button.to_validate': 'Valider mon projet EN',
  'project.reference': 'Reference',
  'project.list.heading.1': 'Good news!',
  'project.list.heading.2': 'Thank you!',
  'project.list.description.1': 'description 1',
  'project.list.description.2': 'description 2',
  'project.send_to_firms': 'Envoyer mon projet aux pros disponibles EN',
  'project.transmitted_to_firms': 'Your project is transmitted to ES',
  'project.question': 'Question & Answers',
  'project.purpose': 'Purpose of the request',
  'project.startTimeframe': 'Project start timeframe',
  'project.startTimeframe.now': 'Now',
  'project.startTimeframe.1_to_3_months': '1 to 3 months',
  'project.startTimeframe.3_to_6_months': '3 to 6 months',
  'project.startTimeframe.6_to_12_months': '6 to 12 months',
  'project.find_a_pro': 'Find a pro to start my project',
  'project.quotation': 'Get a project quotation',
  'project.modify_my_project': 'Modifier mon projet EN',
  'project.continue': 'Continue my project',

  'project.elaboration.submit': 'Submit',
  'project.elaboration.answer': 'tapez votre réponse EN',
  'project.elaboration.title': 'Project Elaboration',
  'project.elaboration.reset': 'Reset',
  'project.elaboration.back': 'Back',
  'project.elaboration.choose_conversation': 'Choose a conversation',
  'project.elaboration.summary.validate': 'Validate your request',
  'project.elaboration.summary.title': 'Vous trouverez ci-joint le résumé de votre demande de projet EN',
  'project.elaboration.project_prevalidation.success': 'The project %name% was created',

  'user.gender': 'Gender',
  'user.forget_password': 'Forget password ?',
  'user.create_account': 'Sign up',
  'user.sign_up': 'Sign up',
  'user.mr': 'Mr',
  'user.mrs': 'Mrs',
  'user.first_name': 'First Name',
  'user.last_name': 'Last Name',
  'user.email': 'Email',
  'user.password': 'Password',
  'user.confirm_password': 'Reenter your password',
  'user.password.show': 'Show / hide password',
  'user.my_account': 'My account',
  'user.my_projects': 'My projects',
  'user.messaging': 'Messaging',
  'user.log_out': 'Log out ➔',
  'user.sign_in': 'Sign in',
  'user.no_account_question': 'No account yet?',
  'user.update': 'Update',
  'user.fixed_phone': 'Phone',
  'user.mobile_phone': 'Mobile phone number',
  'user.postal_code': 'Postal code',
  'user.address': 'Address',
  'user.birthday': 'Birthday',
  'user.facebook_id': 'Facebook id',
  'user.google_id': 'Google id',
  'user.prefered_language': 'Prefered language',
  'user.country_code': 'Country',
  'user.region': 'Region',
  'user.city': 'City',
  'user.send': 'Send',
  'user.resend': 'Renvoyer',
  'user.thank_you': 'Thank you!',
  'user.sign_up.heading': 'Welcome to Quotatis',
  'user.sign_up.description': 'We are looking forward to helping you in the development of your project.',
  'user.sign_up.confirmation': 'An email was sent to confirm your account.',
  'user.reset_password.heading': 'New password',
  'user.reset_password.description': 'Enter your new password to enjoy your personal space.',
  'user.reset_password_email': 'An email was sent to reset your password.',
  'user.reset_password_success': 'Your password has been successfully changed.',
  'user.favorite_page': 'Favorite page',
  'user.project_page': 'My project page',
  'user.message_page': 'My Message page',
  'user.invalid_login': 'Your login and/or password are not valid',
  'user.password_match_err': 'Passwords do not match',
  'user.captcha_error': 'Are you a bot?',
  'user.continue_with_facebook': 'Continue with Facebook',
  'user.continue_with_google': 'Continue with Google',
  'user.newsletter_subscription': 'I want to subscribe to Quotatis weekly newsletter',
  'user.profile_info': 'Profile infos',
  'user.contact_info': 'Contact infos',
  'user.social_network': 'Social Networks Parameters',
  'user.choose_img': 'Choose a picture',
  'user.drop_img': 'Try dropping some files here, or click to select files to upload.',
  'user.forgot_password.heading': 'Forgotten password ?',
  'user.forgot_password.description': 'Please enter your e-mail address below to reset your password.',
  'user.account_updated': 'Your account has been updated',
  'user.coming_back': 'Coming back?',
  'user.code': 'Confirmation code',
  'user.new_sms_sent': 'A new sms has been sent',
  'user.phone_already_verified': 'Your phone number has already been verified',
  'user.send_sms_again': 'Send sms again',
  'user.send_email_again': 'Envoyer à nouveau l\'email EN',
  'user.change_phone_number': 'Change my phone number',
  'user.contactPreference': 'Préférences de contact EN',
  'user.contactPreference.email': 'Par email EN',
  'user.contactPreference.phone': 'Par téléphone EN',
  'user.contactComment.no_preferences': 'No Preference',
  'user.contactComment': 'Jours et Horaires de contact EN',
  'user.contactComment.during_business_hours': 'During business hours',
  'user.contactComment.outside_business_hours': 'Outside business hours',
  'user.phone_validated': 'Your phone number has been verified',
  'user.email.sent': 'A confirmation email has been sent to you',
  'user.email.check': 'Email validation in progress',
  'user.error': 'Oops',
  'user.error.email_not_verified': 'Your email could not be verified',

  'validators.required_field': 'Required field',
  'validators.min_length': 'Must be at least {min} characters',
  'validators.max_length': 'Must be no more than {max} characters',
  'validators.integer': 'Must be an integer',
  'validators.must_match': 'Must match',
  'validators.must_be_one_of': 'Must be one of {values}',
  'validators.user.invalid_email': 'Invalid email address',
  'validators.user.invalid_url': 'Invalid url',
  'validators.user.already_exist': 'This user already exists',
  'validators.user.postalCode.not_blank': 'Field "Postal code" has not to be blank',
  'form.input.button.phone': 'Phone',
  'validators.user.mobilePhone.not_blank': 'Field "Mobile phone" has not to be blank',
  'validators.user.lastName.not_blank': 'Field "Last name" has not to be blank',
  'validators.user.firstName.not_blank': 'Field "First name" has not to be blank',
  'validators.user.gender.not_blank': 'Field "Gender" has not to be blank',
  'validators.user.gender.invalid_choice': 'Field "Gender" is not valid',
  'validators.user.password.not_blank': 'Field "Password" has not to be blank',
  'validators.user.password.must_match_required': 'Your password must contain at least 8 characters & one special char',
  'validators.user.mobilePhone.invalid_type': 'Field "Mobile phone" is not valid',
  'validators.user.fixedPhone.invalid_type': 'Field "Fixe phone" is not valid',
  'validators.forgot_password.unexpired_token': 'You already asked to reinitialize your password, please check your emails.',
  'validators.forgot_password.user_not_found': 'User not found',

  'how_it_works.section_title': 'How it works',
  'how_it_works.describe_project.title': 'Décrivez votre projet EN',
  'how_it_works.describe_project': 'En quelques questions, nous déterminons ensemble les informations nécessaires pour démarrer votre projet de rénovation, construction ou amélioration de votre habitat. EN',
  'how_it_works.verified_pros.title': 'Quotatis sélectionne pour vous jusqu\'à 3 pros vérifiés EN',
  'how_it_works.verified_pros.first_part': 'Nous recherchons autour de vous les pros disponibles EN',
  'how_it_works.verified_pros.link': 'dans notre réseau de professionnels vérifiés EN',
  'how_it_works.verified_pros.second_part': '. Les artisans établissent un devis complet, et vous pouvez lancer les travaux avec celui que vous avez choisi EN',
  'how_it_works.guide.title': 'On vous guide EN',
  'how_it_works.guide.first_part': 'Nous vous accompagnons tout au long de votre projet de travaux avec des  EN',
  'how_it_works.guide.link': 'conseils personnalisés EN',
  'how_it_works.guide.second_part': ' et des guides dédiés. EN',

  'hero.title_message': 'Quels travaux souhaitez-vous faire ? EN',
  'hero.welcome_message': 'Quotatis vous aide à trouver l\'artisan qui concrétisera votre projet. EN',
  'hero.conversation_in_progress': 'Vous avez déjà une conversation en cours. Souhaitez-vous la reprendre ? EN',
  'hero.subheading_in_progress': 'Vous pouvez aussi commencer un nouveau projet! EN',
  'hero.button_message': 'Continuer mon projet EN',

  'thumbnailcard.link': 'Let\'s Go ➔',

  'footer.advices.section_title': 'Nos guides travaux EN',
  'footer.advices.window': 'Fenêtres, Portes & Volets EN',
  'footer.advices.heat': 'Chauffage EN',
  'footer.advices.air': 'Climatisation & Ventilation EN',
  'footer.advices.wall': 'Revêtement des murs EN',
  'footer.advices.ground': 'Revêtement du sol EN',
  'footer.advices.roof': 'Toiture & Façade EN',
  'footer.advices.insulation': 'Isolation & Etanchéité EN',
  'footer.advices.plumbing': 'Plomberie EN',
  'footer.advices.masonry': 'Maçonnerie EN',
  'footer.advices.renovation': 'Rénovation, Construction complète EN',
  'footer.advices.bathroom': 'Salle de bain EN',
  'footer.advices.kitchen': 'Cuisine EN',
  'footer.advices.outside': 'Extérieur EN',
  'footer.advices.wellness': 'Bien-être EN',

  'footer.corporate.language': 'English',
  'footer.corporate.help': 'Help',
  'footer.corporate.about': 'About',
  'footer.corporate.press': 'Press',
  'footer.corporate.phone_help': 'Need help ?',
  'footer.corporate.phone_pros': 'Vous êtes un pro ? EN',
  'footer.corporate.privacy_policy': 'Privacy policy EN',
  'footer.corporate.phone_help_number': '01 77 86 80 97 EN',
  'footer.corporate.phone_pros_number': '01 77 86 71 91 EN',
  'footer.corporate.notice': 'Legal notice',
  'footer.corporate.use': 'Terms & conditions of use',

  'testimonials.section_title': 'Testimonials',
  'testimonials.read_it': 'Read the testimonial ➔',

  'tips_and_tricks.call_to_action': 'Tous nos conseils & astuces EN',
  'tips_and_tricks.section_title': 'Conseils & astuces EN',

  'last_projects.section_title': 'Derniers projets EN',

  'cookies.message': 'En poursuivant votre navigation sur ce site, vous acceptez l’utilisation de cookies pour vous proposer des services et des offres adaptés à vos centres d’intérêts. EN',
  'cookies.more': 'More informations',

  'auto-validation.validate.title': 'Merci d\'avoir validé vos informations EN',
  'auto-validation.validate.message': 'Vous recevrez très prochainement par e-mail les coordonnées des professionnels de confiance que nous aurons sélectionnés pour vous. EN',
  'auto-validation.validate.call_to_action': 'Accéder à mes projets EN',
  'auto-validation.title.informations': 'Vos informations EN',
  'auto-validation.title.contact_preferences': 'Préférences de contact EN',
  'auto-validation.title.project': 'Votre projet EN',

  'auto-validation.mail.title': 'Consultez vos e-mails EN',
  'auto-validation.mail.message': 'Nous venons d\'envoyer un email à l\'adresse ci-dessous. Pour confirmer votre adresse e-mail, veuillez cliquez sur le bouton de confirmation. Cela permettra aux professionnels de vous contacter dès la validation de votre projet. EN',
  'auto-validation.mail.send_mail': 'Renvoyer l\'e-mail EN',

  'auto-validation.phone.title': 'Confirmez votre numéro de téléphone EN',
  'auto-validation.phone.message': 'Ce numéro permettra aux professionnels de vous contacter dès la validation de votre projet. EN',
  'auto-validation.phone.check_phone': 'Confirmer le numéro de téléphone EN',

  'auto-validation.phone_code.title': 'Saisissez le code à 6 chiffres EN',
  'auto-validation.phone_code.message': 'Nous avons envoyé le code à 6 chiffres par SMS au numéro de téléphone indiqué précédemment. Saisissez-le ci-dessous : EN',
}

export default en
