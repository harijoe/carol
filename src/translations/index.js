import messagesEn from './en'
import messagesEs from './es'
import messagesFr from './fr'

const translations = {
  fr: messagesFr,
  en: messagesEn,
  es: messagesEs,
}

export default translations
