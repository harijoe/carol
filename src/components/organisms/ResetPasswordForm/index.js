import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import injectTranslate from 'i18n/hoc/injectTranslate'
import { theme, breakpoint } from 'utils/style'

import { Button, CarouselPageTemplate, AnimatedLabelField } from 'components'

const Form = styled.form`
  width: 100%;
  box-sizing: border-box;
  
  ${breakpoint('m')`
    width: 50%;
  `}

  strong {
    color: ${theme('colors.danger')};
  }
`

const StyledButton = styled(Button)`
  margin-top: ${theme('spaces.xl')};

  ${breakpoint('m')`
    margin-top: ${theme('spaces.xxl')};
  `}
`

const ResetPasswordForm = ({ error, handleSubmit, loading, translate }) => (
  <CarouselPageTemplate
    heading={translate('user.reset_password.heading')}
    description={translate('user.reset_password.description')}
  >
    <Form onSubmit={handleSubmit}>
      <Field name="token" type="hidden" component="input" />
      {error && <FormattedMessage id={error} tagName="strong" />}
      <AnimatedLabelField
        name="password"
        type="password"
        icon="pwd-login"
        label={translate('user.password')}
      />
      <AnimatedLabelField
        name="passwordValidation"
        type="password"
        icon="pwd-login"
        label={translate('user.confirm_password')}
      />
      <StyledButton type="submit" loading={loading}>
        <FormattedMessage id="user.send" />
      </StyledButton>
    </Form>
  </CarouselPageTemplate>
)

ResetPasswordForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.string,
  translate: PropTypes.func.isRequired,
}

export default injectTranslate(ResetPasswordForm)
