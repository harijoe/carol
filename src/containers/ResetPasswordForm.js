import React from 'react'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createValidator, required, match, password } from 'services/validation'
import { resetPassword } from 'store/actions'
import { fromStatus } from 'store/selectors'

import { ResetPasswordForm } from 'components'

const ResetPasswordFormContainer = props => (
  <ResetPasswordForm {...props} />
)

const onSubmit = (data, dispatch) => dispatch(resetPassword.request(data, data.token))

const validate = createValidator({
  password: [password],
  passwordValidation: [required, match('password')],
})

const mapStateToProps = state => ({
  loading: fromStatus.getLoading(state).USER_UPDATE_PASSWORD,
  initialValues: {
    token: typeof state.routing !== 'undefined' ? state.routing.locationBeforeTransitions.query.token : '',
  },
})

export const config = {
  form: 'ResetPasswordForm',
  fields: ['password', 'passwordValidation'],
  destroyOnUnmount: false,
  onSubmit,
  validate,
}

export default connect(mapStateToProps)(reduxForm(config)(ResetPasswordFormContainer))
