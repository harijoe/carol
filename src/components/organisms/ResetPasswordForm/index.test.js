import React from 'react'
import { shallow } from 'enzyme'

import { messages } from '../../../translations/en'
import mockIntl from '../../../../test/intlMock'
import ResetPasswordForm from './'

it('renders ResetPasswordForm', () => {
  const wrapper = shallow(mockIntl(<ResetPasswordForm />))

  expect(wrapper).toMatchSnapshot()
})
