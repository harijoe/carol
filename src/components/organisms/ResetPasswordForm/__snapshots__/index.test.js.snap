exports[`test renders ResetPasswordForm 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <InjectIntl(ResetPasswordForm) />,
  "nodes": Array [
    <InjectIntl(ResetPasswordForm) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <IntlProvider
        locale="en"
        messages={Object {}}>
        <InjectIntl(ResetPasswordForm) />
</IntlProvider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": IntlProvider {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "intl": undefined,
        },
        "props": Object {
          "children": <InjectIntl(ResetPasswordForm) />,
          "locale": "en",
          "messages": Object {},
        },
        "refs": Object {},
        "state": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
          "now": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <InjectIntl(ResetPasswordForm) />,
        "_debugID": 2,
        "_renderedOutput": <InjectIntl(ResetPasswordForm) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <IntlProvider
    locale="en"
    messages={Object {}}>
    <InjectIntl(ResetPasswordForm) />
</IntlProvider>,
}
`;
